Challenge-based competition (#42)
robocup-logistics/rcll-rulebook: challenges.tex

\usepackage[utf8]{inputenc}
\input{macros.tex}

\newcolumntype{C}{>{\raggedright\arraybackslash}X}

\newcommand{\refsec}[1]{Section~\ref{#1}}
\newcommand{\reffig}[1]{Figure~\ref{#1}}
\newcommand{\refdef}[1]{Definition~\ref{#1}}
%\newcommand{\reflst}[1]{Listing~\ref{#1}}
\newcommand{\reflst}[1]{Figure~\ref{#1}}
\newcommand{\reftab}[1]{Table~\ref{#1}}
\usepackage{acronym}
\acrodef{BS}{base station}
\acrodef{CS}{cap station}
\acrodef{RS}{ring station}
\acrodef{SS}{storage station}
\acrodef{DS}{delivery station}
\acrodef{MPS}{modular production system}
\acrodef{RCLL}{RoboCup Logistics Leaugue}
\acrodef{TBD}{To be Discussed}

\usepackage{todonotes}
\presetkeys%
    {todonotes}%
    {inline}{}
\begin{document}
\begin{titlepage}
 \vspace*{5cm}
 \begin{center}
  \begin{LARGE}

   {\bf RoboCup Logistics League}\\[2ex]
   {\Large \bf{Challenges}}\\[4ex]
   {\Large Rules and Regulations 2021}\\[4ex]
  \end{LARGE}
  \hrule

  {\LARGE\vspace*{4ex}}
  \begin{Large}
   The Technical Committee 2012--2021\\[6ex]
  \end{Large}
  \begin{tabular}{lll}
   \textbf{Vincent Coelen}&\textbf{Christian Deppe}&\textbf{Mostafa Gomaa}\\
   \textbf{Till Hofmann}&\textbf{Ulrich Karras}&\textbf{Tim Niemueller}\\

>    \textbf{Alain Rohr}&\textbf{Thomas Ulz}&\textbf{Tarik Viehmann}\\[.5em]

Reviewer: Guess we have to update the names of the TC Members.

around the insertion zone and robots may be move freely in and out of the
insertion zone.
\subsection{Mockup Machines}\label{sec:machines}
In case no real \ac{MPS} stations are available, replications
(so called \emph{mockup machines}) may be used, that do not need to
physically perform the respective production steps. Instead, that work may
be carried out by a human supervisor (see \refsec{sec:operators}).
The minimum requirements for a mockup machine are specified in the following.

Mockup machines are required to have the same box-like base-shape as specified
in the RCLL rulebook.
Additionally, the following parts need to be mounted:
\begin{itemize}
	\item a model of the conveyor belt
	\item a shelf on the front right side of the box on stations replicating a \ac{CS}
	\item either a shelf, a slide or a conveyor belt on the front right side
of the box, such that it is accessible from the front on stations replicating a \ac{RS}
\end{itemize}
Models for a conveyor belt, shelf and slides can be found in the RCLL rulebook
repository\footnote{\url{https://github.com/robocup-logistics/rcll-rulebook/tree/common/challenges/mock_up_models}}.

> The building materials for the models must be opaque, but may have any color.

Reviewer: Hos close do the mock up conveyor/shelf/slide have to be to that model? Since it is not possible to 3D print them as a whole. Must it be exact or are those files just meant to be an orientation help?

Reviewer 2: Since it is hard to exactly determine how close your model is to the real thing, as long as it looks like a slide, conveyor, shelf on the recording it will be fine.
If you need a precise model for your own software and hardware setup you free to make it as precise as you want.

Reviewer: ok, thanks.

In order to compete in all main challenges, a minimum number of 3 mockup
machines are required. On higher difficulty some challenges may require
4 mockup stations and some secondary challenges require 7 stations.

\subsection{Remote Setup}
In case a competition is carried out remotely, a proper local setup has to
be established and approved by the OC.
Requirements include a proper camera setup that covers the field sufficiently,
such that external viewers can verify the integrity of each challenges,
as well as an approval for every mockup machine and robot that is used.
After registration the OC will verify the field setup in a video call.
The validation call will be scheduled individually for each team
to account for timezones.

\section{Game Play}
\subsection{Competition Scope}
Each team gets slots assigned to participate in challenges.
Each slot can be used to do at most one challenge and is split in two phases:

> 1. In the \texttt{SETUP} phase the participants ensure that

Reviewer: How many slots does a team get? One for each possible challange,or will it be more/less? Also is it possible that slots of 2 teams overlap?

Reviewer 2: The competition is asynchronous. There will probably be overlapping times, but the overhead of coordinating is too large.
Slot in this context describes a 30 minute time window where you can fulfill the challenge. Nonetheless, the challenge should be streamed and recorded.
The number of slots is theoretically unlimited, technically you get 48 slots per day.

Reviewer: On another comment @TarikViehmann  said: That depends on the OC as they are responsible to come up with the organization of the timeslots.

So are there now unlimited numbers of slots, or will the OC decide this?

Reviewer 2: The slots can not be limited, due to different access times and timezones and so on.
So we split up the day into 48 slots of 30 minutes (meaning you can not create an artificial 49th slot by finishing a challenge earlier). In the very beginning there was a thought to provide teams with a fixed amount of slots (like every team gets 20 slots), we disregarded that due to organizational overhead, etc.

\begin{itemize}

> 	\item the video stream is setup,

Reviewer: Should we fix a software to use here or will it be provided by the RoboCup federation? Not that every team comes up with a different application to stream their games.

Reviewer 2: I think we agreed on Zoom to record and stream.
And there will be a inspection of the recording setup of each team in the first week of June (once all teams are registered).

Once the competition is finished, $5$ bonus points are awarded each time a
team solved a challenge on a difficulty in the shortest amount of time.

\subsubsection{Navigation Challenge}\label{sec:challenge-navigation}
Basic navigation task with known obstacles.\\
\textbf{Task:} Drive to 12 randomly generated target zones. A target zone is
reached, if any robot remains in that zone for at least 5 consecutive seconds.
The pose of each robot must be set in the beacon message
for the RefBox to properly register the zones as visited. The robot may move
within the zone during the 5 second period.\\
Variations of this challenge depend on the number of available machines
(see \reftab{tab:challenge-navigation}).
Multiple robots may be used to simultaneously to reach multiple target zones.
Partial points may be awarded in case only a subset of target zones were
reached.

\begin{table}[!htb]
 \centering
 \begin{tabular}{l|l|l|l|l}
  \multirow{2}{*}{Machines}
  & \multicolumn{4}{c}{Scoring} \\\cline{2-5}
	& $\geq$ 4 zones  & $\geq$ 8 zones & 12 zones  & max \\\hline\hline
	 2 & +2 & +4 & +9 & 15 \\
	 3 & +3 & +6 & +14 & 23 \\
	 4 & +4 & +8 & +18 & 30 \\

>  \end{tabular}

Reviewer: I don't really understand the `max` here. Above it is stated that its 12 zones, so what is the difference between 12 and max?

Reviewer: ok now I understand, once you reach 4 you get 2 point, then you get 4 points if you reach 8 and then finally 9 points once you visit all 12 zone, thus the sum is 15, took me some while to get that :D

Author: i mention that explicitly now, i hope that avoids confusion

	\label{fig:grasping-field}
\end{figure}

\subsubsection{Product Challenges}\label{sec:challenge-cx}
This section covers four types of challenges, instead of just a single one.
Each challenge corresponds to the production of a product with one of the
available complexities (C0, C1, C2, C3) in the \ac{RCLL} using either
one or two \ac{RS}.\\
For complexities C1, C2 and C3 the accumulated cost for mounting the required
rings must be equal to 1, 2 and 3, respectively.
\textbf{Task:} Produce all posted orders.\\
\begin{table}[!htb]
 \centering
 \begin{tabular}{l|l|l}
  Machines & Challenge type & Scoring \\\hline
  2 & C0 & 30 \\
  3 & C1 & 50 \\
  4 & C1 & 50 \\
  3 & C2 & 70 \\
  4 & C2 & 70 \\
  3 & C3 & 100 \\
  4 & C3 & 100 \\
 \end{tabular}
 \caption{CX Challenge}
 \label{tab:challenge-cx}
\end{table}

\subsubsection{Exploration + C0 Challenge}\label{sec:challenge-combine-exp-c0}
A simple production task on a field with unknown machine positions.
The challenge is to produce a product of complexity C0 without receiving the
machine positions at the start of the production phase, resembling a unified
exploration and production phase.\\
\textbf{Task:} Produce all posted orders.\\

> Beating this challenges yields 50 points.

Reviewer: Since it is stated above, I guess this here explains `Exploration + C0`,  `Exploration + C1`,  `Exploration + C2`, `Exploration + C3` and it is possible for a team to play all 4 challanges or does a team has to select one out of those?

Reviewer: Ok my fault, seems like in Product Challange positions are published by the refbox, and the combination (Exploration + Production) is only available for C0. I would make that also 4 challanges.

Author: Yea the main reason for limiting exploration+Production to a single challenge is to not give it so much impact for the end result as its a new concept. Also there was the concern of having too many challenges to compete in (the idea for the normal production with machine positions is indeed that you can play all 4 of them). That depends on the OC as they are responsible to come up with the organization of the timeslots.

I personally like the concept of Exploration+Production and would not be opposed to have more challenges in that area but i am not sure if it would spread the gap between advanced teams and newer ones too much.

We could add those additional complexities for Exploration+Production extra challenges under the Secondary Challenge category that does not yield any points towards the actual challenge and is just for fun.

Reviewer: I think that since the exploration is already part of the game for a long time it is not  new concept and hence would like to see it in the Primary Challanges, how do the other people see this? since there is not a big difference between the Production / Production + Exploration challange in my eyes, as long as you are capable of playing the exploraion phase in RCLL.

For me it is also no where stated that a team has to compete in every challange, so I don't see a problem with more challanges that slots. Like if there are 15 challanges I can select which 10 I want to do. If I select one which yields more points the risk that I fail is higher.

Regarding your thought of the gap, I am well aware that the league is hard to join, but limiting the possibilities for better teams will just result in frustration in the long run.

Reviewer 2: But that is something (among other things that will be clarified as part of the initial start of league meeting, since it is not valid for the challenges to come)

Reviewer 2: Since the secondary challenge is scored separately I don't see any issue with you playing on a smaller field with less machines in this context. (I share Tariks concerns of inflating the potential points teams can score with this challenge)

Author: i agree with Peter that it makes sense to set an upper limit regardless. big teams with the capacity to run like 20 slots a day have a huge advantage over smaller teams or those with tighter time constraints. Setting some limit might be a good idea. If 10 works we can do that.

Regarding the secondary challenge stuff: fair point so lets roll with it and put more challenges in the main competition. I will update this. When noone else gives feedback noone can complain :P

Regarding changes to the real game: what about scheduling a TC meeting to discuss it with all interested ones to push rule changes on the way? I can organize that and then we go from there.

Author: yea i would prefer to shift TC decision discussions to github as it is way easier to manage this.
I need to dig up the request to answer it. For me there is also no problem with allowing that arm ore abolishing the height restriction altogether.

%$10$ points
%\begin{table}[!htb]
% \centering
% \begin{tabular}{l|l|l|l}
%  Points & Scoring \\\hline
%  $[0,50)$ & 0\\% chktex 9
%  $[50,150)$ & 20\\% chktex 9
%  $[150,250)$ & 40 \\% chktex 9
%  $[250,\infty)$ & 60 \\% chktex 9
% \end{tabular}
% \caption{CX Challenge}
% \label{tab:challenge-cx}
%\end{table}

\subsubsection{Markerless Detection Challenge}\label{sec:markerless}
Image recognition challenge to classify different machine types.\\
\textbf{Task:} Autonomously label the machines shown in a set of pictures,
a picture may contain more than one machine, which all count towards the
total number of machines to detect.\\
As a preparation for this challenge, a data set will be supplied to
all participants which may be used for training and testing purposes
\footnote{\url{https://fh-aachen.sciebo.de/s/MRghypMQSMhTYaC}}.
The evaluation set for the challenge consists of a set of separate images.
\begin{table}[!htb]
 \centering
 \begin{tabularx}{\linewidth}{l|l|l|l}
  \% Correctly Classified & \% Wrongly Classified & \% Not Classified
  & Scoring \\\hline
  $x$ & $y$ & $z$ & $(x-y)\cdot30$
 \end{tabularx}

>  \caption{Machine Detection Challenge}

Reviewer: where is the image set? In the rulebook repo?

Author: the link is right there:
```
\footnote{\url{https://fh-aachen.sciebo.de/s/MRghypMQSMhTYaC}}.
```

\acrodef{RS}{ring station}
\acrodef{SS}{storage station}
\acrodef{DS}{delivery station}
\acrodef{MPS}{modular production system}
\acrodef{RCLL}{RoboCup Logistics Leaugue}
\acrodef{OC}{Organizational Committee}
\acrodef{TBD}{To be Discussed}

\usepackage{todonotes}
\presetkeys%
    {todonotes}%
    {inline}{}
\begin{document}
\begin{titlepage}
 \vspace*{5cm}
 \begin{center}
  \begin{LARGE}

   {\bf RoboCup Logistics League}\\[2ex]
   {\Large \bf{Challenges}}\\[4ex]
   {\Large Rules and Regulations 2021}\\[4ex]
  \end{LARGE}
  \hrule

  {\LARGE\vspace*{4ex}}
  \begin{Large}
   The Technical Committee 2012--2021\\[6ex]
  \end{Large}
  \begin{tabular}{lll}
   \textbf{Vincent Coelen}&\textbf{Christian Deppe}&\textbf{Till Hofmann}\\
	\textbf{Ulrich Karras}&\textbf{Peter Kohout}&\textbf{Alain Rohr}\\
	\textbf{Matthias Studer} &\textbf{Thomas Ulz}&\textbf{Tarik Viehmann}\\[.5em]
   Daniel Ewert&Nils Harder&S\"oren Jentzsch\\
   Nicolas Meier&Sebastian Reuter&Wataru Uemura\\
   Gerald Steinbauer&Tobias Neumann&Mostafa Gomaa\\
	 Tim Niemueller
  \end{tabular}
  \vfill
  Revision Date: 07.05.2021\\
 \end{center}
\end{titlepage}

\section{Introduction}
\label{sec:intro}

> This additional rulebook aims to provide a competition in the RoboCup logistics

Reviewer 3: What about
```suggestion
This rulebook addendum provides a competition in the RoboCup logistics
```

league domain, where the main tasks of the league are captured in isolated
challenges.
The objectives of this new competition are:
\begin{itemize}
 \item to provide a framework that allows teams to show and evaluate their
       progress in the individual tasks of the RCLL
 \item to ease the preparation for the main competition through providing a
       simplified cost- and space-efficient setup suitable for replication in
       local labs
 \item to be attractive for both RoboCup live events and online competitions,
       where teams can participate remotely from all over the world
\end{itemize}
This document is not self-sufficient, instead all the rules from the main
rulebook\footnote{\url{https://github.com/robocup-logistics/rcll-rulebook/releases}}
apply, unless stated otherwise.
The proposed format is carried out at RoboCup 2021, which is held remotely.
Hence the remainder of this document establishes the rules around the online
competition.

\section{Qualification}
\label{sec:qualification}
In addition to the regular Qualification process, teams have to participate
in a \href{https://fh-aachen.sciebo.de/s/Qkm7VfIIEIIJk1d}{RefBox workshop, see here}.

> To complete the workshop, a game report dump of at least 2 Challenges

Reviewer 3: nit:
```suggestion
To complete the workshop, a game report dump of at least two challenges
```

has to be submitted to the \acf{OC}.

\section{Requirements to score points}
\label{sec:scoring}
For your challenges to count as completed and score points you need to submit
the recorded video footage to the OC.
Further you need to submit your game reports from the MongoDB to the OC.

> Your submission will not be counted if the footage shows human

Reviewer 3: An `enumerate` with the two items may make this slightly easier to read?

interference with the field, machines or robots outside of
the specified rules in Section \ref{sec:operators}.

\section{Competition Area}
\subsection{Field Layout}
The competition area for the main challenges consists of a \SI{5 x 5}{\metre}
area divided in square zones of \SI{1 x 1}{\metre}. Additional challenges that
are not counting towards the scoring of the competition are carried out on

> a \SI{7 x 8}{\metre} field instead (corresponding to halve the field of the

Reviewer 3: ```suggestion
a \SI{7 x 8}{\metre} field instead (corresponding to half the field of the
```

regular \ac{RCLL} field).

> The field is a subset of the magenta-halve of the regular field as depicted

Reviewer 3: ```suggestion
The field is a subset of the magenta half of the regular field as depicted
```

	\node at (5,5) [circle,fill,inner sep=1.5pt,label={[font=\scriptsize]135:$(-5,5)$}]{};
	\node at (0,5) [circle,fill,inner sep=1.5pt,label={[font=\scriptsize]45:$(0,5)$}]{};
	\draw[line width=2pt,red,->] (0,0) -- node[above left,pos=1] {$x$} (-1.5,0);
	\draw[line width=2pt,red,->] (0,0) -- node[below left,pos=1] {$y$} (0,1.5);
\end{tikzpicture}
	\caption{The \SI{5 x 5}{\metre} playing field.}
	\label{fig:challenge-field}
\end{figure}

The entire area belongs to a single team. The bottom left-most
\SI{1 x 3}{\metre} area is called the insertion zone and provides the starting
positions for up to three robots.
Contrary to the full RCLL field, it is not necessary to build partial walls
around the insertion zone and robots may be move freely in and out of the
insertion zone.
\subsection{Mockup Machines}\label{sec:machines}
In case no real \ac{MPS} stations are available, replications
(so called \emph{mockup machines}) may be used, that do not need to
physically perform the respective production steps. Instead, that work may
be carried out by a human supervisor (see \refsec{sec:operators}).
The minimum requirements for a mockup machine are specified in the following.

> Mockup machines are required to have the same box-like base-shape as specified

Reviewer 3: nit:
```suggestion
Mockup machines are required to have the same box-like base shape as specified
```

in the RCLL rulebook.
Additionally, the following parts need to be mounted:
\begin{itemize}
	\item a model of the conveyor belt
	\item a shelf on the front right side of the box on stations replicating a \ac{CS}
	\item either a shelf, a slide or a conveyor belt on the front right side
of the box, such that it is accessible from the front on stations replicating a \ac{RS}
\end{itemize}
Models for a conveyor belt, shelf and slides can be found in the RCLL rulebook
repository\footnote{\url{https://github.com/robocup-logistics/rcll-rulebook/tree/common/challenges/mock_up_models}}.

The building materials for the models must be opaque, but may have any color.

In order to compete in all main challenges, a minimum number of 3 mockup
machines are required. On higher difficulty some challenges may require
4 mockup stations and some secondary challenges require 7 stations.

> 

Reviewer: Good question. I would like to have it as close to the real game as possible, is there data on how long the machines need to produce a piece? Is it easy to notice in the WebUi of the refbox that a machine has been prepared? I did not check that the last time I used it. 
It might be hard to control if teams wait the right amount of time before picking it up. I guess doing a immediate processing make it more easy. But if the refbox is waiting already the right amount of time till it sets the machien into ReadyAtOutput I think as long as it is good communicated to the teams (via the UI) it should be doable.

> 

Reviewer 2: The RefBox has an internal databases of approximated processing times, so that in mockup mode the machines do not finish immediately.
The UI does show when a machine is ready or processing. 
So in theory no robot should pick the product up before the machine finished and the refbox sent the signal (regardless of when an operator moved the workpiece), if you pick it up before the refbox sent the signal, then there is something wrong in your world model.

\subsection{Remote Setup}
In case a competition is carried out remotely, a proper local setup has to
be established and approved by the \ac{OC}.
Requirements include a proper camera setup that covers the field sufficiently,
such that external viewers can verify the integrity of each challenges,
as well as an approval for every mockup machine and robot that is used.
After registration the \ac{OC} will verify the field setup in a video call.
The validation call will be scheduled individually for each team

> to account for timezones.

Reviewer 3: It would make sense to be more precise on what exactly is expected and not leave it to the OC to freely decide whether the setup is suitable.

\section{Game Play}
\subsection{Competition Scope}
Each team can attempt a challenge by specifying a 30 minute time slot along
with the challenge in advance of the respective competition day.
The announcement procedure is managed by the \ac{OC}.
Each slot can be used to do at most one challenge and is split in two phases:

> 1. In the \texttt{SETUP} phase the participants ensure that

Reviewer 3: `enumerate`?

\begin{itemize}
	\item the video stream is setup via Zoom,
	\item the RefBox is started with the appropriate settings for the challenge,
	\item the machines and robots are placed on the field according to the rules
	\item and the robots are started.
\end{itemize}
The setup must be completed within 10 minutes, otherwise the attempt is counted
as a failure.
Then the \texttt{PRODUCTION} phase starts by instructing the RefBox accordingly.
During the production phase the team has 20 minutes to solve the challenge.

A team may retry the challenge by setting the RefBox back to the setup phase

> and then starting the production phase again.

Reviewer 3: What happens to the machines, robots, will everything be reset? This should be stated explicitly.

Only the last attempt in each slot can be counted. A team may decide to count
no attempt at all. Once a challenge is counted, it
may not be retried again, unless the difficulty is increased.
In order to complete an attempt, the RefBox needs to be set to phase
\texttt{POST\_GAME}.

All challenges are conducted while measuring the execution time of the
corresponding attempt, starting from the begin of the \texttt{PRODUCTION} phase
and ending at the begin of \texttt{POST\_GAME}.
The fastest team to complete a challenge on the selected difficulty gains $5$
additional points. A challenge is only completed if the full score is achieved,
so getting only partial credits for a challenge disqualifies from getting the
bonus points for the fastest completion.

\subsection{Changes compared to the Main Competition}

> The tasks covered in the various challenges mostly obey the regular rules

Reviewer 3: nit: a task can't really obey rules

  & Color  & Price & Color  & Price & \multicolumn{2}{c}{}\\\cline{2-7}
  & Yellow & 0 & Green & 0
  & RS1: $\colconfig_1$ & RS1: $\colconfig_2$ \\
  & Blue  & 1 & Orange & 2
  & RS2: $\colconfig_2$ & RS2: $\colconfig_1$ \\\hline\hline
  Configuration & \multicolumn{2}{c||}{$\colconfig_1$}
  & \multicolumn{2}{c||}{$\colconfig_2$}
  & $\texttt{option1}$ & $\texttt{option2}$\\
 \end{tabular}
 \caption{Materials}
 \label{tab:ring-costs}
\end{table}

\paragraph{Materials}\label{sec:materials}
The available material that can be used per challenge is restricted
(unless stated otherwise) per machine according to the information in
\reftab{tab:materials}.
\begin{table}[!htb]
 \centering
  \begin{tabularx}{\linewidth}{l|l}
   Machine & Available Material  \\\hline
   \ac{BS} & 2 bases of each color \\
   \ac{CS} & 3 cap-carriers (cap color choices up to each team)  \\
   \ac{RS} & 4 rings of each assigned color (8 in total)  \\
  \end{tabularx}
 \caption{Materials}
 \label{tab:materials}
\end{table}

\paragraph{Orders}
Unless specified otherwise, orders that have to be fulfilled in challenges
are entered through the web shop
(see the RefBox workshop in \refsec{sec:qualification})

> by any member of the competing team.

Reviewer 3: What is the competing team? IIUC, there is not (one) opponent, because each team runs the challenges by itself. So who exactly can do this?

regular RCLL rules, those points do not count towards this competition.
Instead, the point scoring for each challenge is listed in
\refsec{sec:challenges}.

\subsection{Field Operators}\label{sec:operators}
In challenges where mockup machines are used, the actual assembly steps have
to be performed by human supervisors. Whenever a machine is instructed,
the RefBox operator announces the required interaction. One field operator may
proceed to enter the field in order to perform the interaction. Afterwards the
field has to be left immediately.
The usual rules for replenishing resources (respecting the limited materials
\refsec{sec:materials}) apply.

\subsection{Available Challenges for the Primary Competition}
\label{sec:challenges}
Challenges have different types and variations (difficulty levels).
The overall score of the competition is calculated by summing up the score
in the highest difficulty achieved in each of the challenge types.
The challenge types of the competition are described in
\refsec{sec:challenge-navigation}-\ref{sec:challenge-markerless}.

The RefBox is used to log the progress and data for each challenge.

> Participating teams are required to send the corresponding game-report dumps
> to the \ac{OC}.

Reviewer 3: It would be helpful to have more information (in the rulebook addendum) on how to create the game report.

nit:
```suggestion
Participating teams are required to send the corresponding game report dumps
to the \ac{OC}.
```

Author: This is covered by the refbox tutorial video

%that can be executed by sending dedicated commands to the RefBox via protobuf.
%Hence no actual robot is required to participate.
%\textbf{Task:} Play a regular RCLL game through the RefBox simulation
%interface.\\
%Participating in this challenge yields points based on the achieved in-game
%points, see \reftab{tab:challenge-simulation}.
%Additionally, the team scoring the highest points overall gets awarded another
%$10$ points
%\begin{table}[!htb]
% \centering
% \begin{tabular}{l|l|l|l}
%  Points & Scoring \\\hline
%  $[0,50)$ & 0\\% chktex 9
%  $[50,150)$ & 20\\% chktex 9
%  $[150,250)$ & 40 \\% chktex 9
%  $[250,\infty)$ & 60 \\% chktex 9
% \end{tabular}
% \caption{CX Challenge}
% \label{tab:challenge-cx}
%\end{table}

\subsubsection{Markerless Detection Challenge}\label{sec:markerless}
Image recognition challenge to classify different machine types.\\
\textbf{Task:} Autonomously label the machines shown in a set of pictures
by their type.
A picture may contain more than one machine, which all count towards the
total number of machines to detect.\\
As a preparation for this challenge, a data set will be supplied to

> all participants which may be used for training and testing purposes

Reviewer: Till now I thought that it is allowed to train only with the supplied data set. Since most of the teams (including GRIPS) do not have access to real machines to create training data I think this is a quite large handicap in this challange if other teams can easily create more training data. How are you seeing this?

Reviewer 2: I totally agree, the idea was to leave this open so teams can come up with additional creative ways to generate data (solidus thought about using images from the simulation).
At Carologistics we are aware of the advantage we have with access to the machines, which is why we only used the published data set for training.
Additionally, I don't know how big the contribution of other data might be because the images for the challenge will be taken from the same machines as the ones in the data set.

\footnote{\url{https://fh-aachen.sciebo.de/s/MRghypMQSMhTYaC}}.
The evaluation set for the challenge consists of a set of separate images.
\begin{table}[!htb]
 \centering
 \begin{tabularx}{\linewidth}{l|l|l|l}
  \% Correctly Classified & \% Wrongly Classified & \% Not Classified
  & Scoring \\\hline

>   $x$ & $y$ & $z$ & $(x-y)\cdot30$

Reviewer 3: So the maximum score is 3000 points? That's a lot

Reviewer: Isn't it less, I think it not possible to distinguish Cyan from Magenta machines, and 1 and 2 (RS1 vs. RS2). Which leads to BS, CS, RS, DS, SS so i guess maximum is 5 * 30 = 150? But since producing a C3 is only worth 100 points, 120 if it is done without maschine positions. 150 is still a lot of points (most that any challange gives). Do we want to push teams to put effort into this over other challanges? If not I would suggest lowering the points, in my opinion other challanges are harder to achieve.

Reviewer 2: I believe the number 30 as just an arbitrary number and I agree that it should be lower. I think a max of 50-70 points might be a good way to go.

Author: It was intended to bring 30 points total across all data to be labeled (independent of the types). I clarified that.

 \end{tabularx}
 \caption{Machine Detection Challenge}
 \label{tab:challenge-markerless}
\end{table}

\subsection{Challenges for the Secondary Competition}

\subsubsection{Full game}\label{sec:challenge-full-game}
Play a full RCLL game on a field of \SI{7 x 8}{\metre} with $7$ machines

> (no machines from the opposing team). The normal RCLL rules apply.

Reviewer 3: Assumingly with all the changes from 5.2?
